### PR TITLE
Adds a default key generation transparent to the user

### PR DESCRIPTION
**Pull request by sr-gi**
Makes the key generation process transparent to the user to improve the UX.

Making the user perform the key generation on bootstrap was a bit odd and made deployment more difficult.

The user is now required to add the `tower_id` as a parameter to the register method. Given that the client is sort of a testing client, only a single tower can be used at the same time (this remains the same as before).

Furthermore, keys are automatically generated for the `cli` and the `tower` on the first use. The tower can request a key overwrite using `--overwritekey` in case the identity needs to be changed by some reason. As a followup, having an import modifier to allow importing external keys may be nice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ and / or
 
 ### Tower id and signing key
 
-`teos` needs a pair of keys that will serve as tower id and signing key. The former can be used by users to identify the tower, whereas the latter is used by the tower to sign responses. You can follow [generate keys](#generate-keys) to generate them.
+`teos` needs a pair of keys that will serve as tower id and signing key. The former can be used by users to identify the tower, whereas the latter is used by the tower to sign responses. This keys are automatically generated on the first run, and can be refreshed by running `teos` with the `--overwritekey` flag.
```
Comment (bigspider) [inline README.md:49]: Typo "these keys"

File: cli/teos_cli.py [hunk 2]
```diff
@@ -228,35 +229,24 @@ def get_all_appointments(teos_url):
...
         user_sk_path (:obj:`str`): path to the user's private key file.
 
     Returns:
-        :obj:`tuple`: a three-item tuple containing a ``str``, a ``PrivateKey`` and a ``str``
-        representing the tower id (compressed pk), user sk and user id (compressed pk) respectively.
+        :obj:`tuple`: a tuple containing a ``PrivateKey`` and a ``str`` representing the user sk and user id
+        (compressed pk) respectively.
 
     Raises:
         :obj:`InvalidKey <cli.exceptions.InvalidKey>`: if any of the keys is invalid or cannot be loaded.
```
Comment (bigspider) [inline cli/teos_cli.py:244]: Remove `any of` from the docs, as there's now only one key in this function.

File: cli/teos_cli.py [hunk 3]
```diff
@@ -270,7 +260,33 @@ def load_keys(teos_pk_path, user_sk_path):
...
+    Loads the tower id from disk.
+
+    Args:
+        teos_pk_path (:obj:`str`): path to the tower's public key file.
+
+    Returns:
+        :obj:`str`: The tower id.
+
+    Raises:
+        :obj:`InvalidKey <cli.exceptions.InvalidKey>`: if any the key is invalid or cannot be loaded.
```
Comment (bigspider) [inline cli/teos_cli.py]: Remove `any`.

File: test/common/unit/test_cryptographer.py [hunk 4]
```diff
@@ -101,13 +99,53 @@ def test_decrypt():
     assert Cryptographer.decrypt(encrypted_data, key) == data
 
 
+def test_generate_key():
+    # Not much to test here in the current approach, this simply shadows PrivateKey()
+    assert isinstance(Cryptographer.generate_key(), PrivateKey)
+
+    # Making sure multiple calls to the function do not return the same key
+    issued_keys = []
```
Comment (bigspider) [inline test/common/unit/test_cryptographer.py:107]: Might be better to make `issued_keys` a `set`.

**Review by bigspider — CHANGES_REQUESTED** [review]
Few minor things only.

File: cli/teos_cli.py (hunk 2 above)
Comment (sr-gi) [reply cli/teos_cli.py:244]: Actually it loads both the private and public key, so any applies here.

File: test/common/unit/test_cryptographer.py (hunk 4 above)
Comment (sr-gi) [reply test/common/unit/test_cryptographer.py:107]: Not a big deal given it is only a test.

File: cli/teos_cli.py (hunk 3 above)
Comment (sr-gi) [reply cli/teos_cli.py]: Fixed in fdef20f

File: README.md (hunk 1 above)
Comment (sr-gi) [reply README.md:49]: Fixed in fdef20f

**Pull request merged**